clang/test/CodeGenSYCL/loop_unroll.cpp

```diff
@@ -1,7 +1,9 @@
-// RUN: %clang_cc1 -triple spir64-unknown-unknown -disable-llvm-passes -fsycl-is-device -emit-llvm %s -o - | FileCheck %s
+// RUN: %clang_cc1 -triple spir64-unknown-unknown -disable-llvm-passes -fsycl-is-device -fgpu-rdc -emit-llvm %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-RDC
+// RUN: %clang_cc1 -triple spir64-unknown-unknown -disable-llvm-passes -fsycl-is-device -fno-gpu-rdc -emit-llvm %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-NORDC
```

Comment:
I don't think it's a good idea to extend all failing tests with `-fno-gpu-rdc` case. For example, the purpose of this test is to check llvm loop metadata and we don't want this test to fail on changing linkage type. The right fix for this test is to drop checking linkage type in `CHECK-LABEL:` patterns.

I would expect 1-2 tests to check the linkage type, but all other tests should ignore it.

Reply:
Note though that the test was already checking the linkage type, that is not new behaviour of the test. When something like that changes, is it more common to just update the test to whatever is generated by the new version, or is it more common to update the test to allow any result? In upstream LLVM, it would be the former, so I went with that approach. If DPC++ usually does it differently, happy to make that change.

I can definitely agree that this is not what this particular test is about, and see your point that checking both `-fgpu-rdc` and `-fno-gpu-rdc` is overkill. Going over the tests, I think for `device_global.cpp` and `sycl-cuda-host-device-functions.cu`, checking both `-fgpu-rdc` and `-fno-gpu-rdc` makes sense for the test, and if we have those two, we cover both functions and variables. For the others, I will take out the `-fno-gpu-rdc` versions of the tests, and then check with you once that is ready if you would still like me to additionally loosen the checks.

Reply:
Updated. Looking over the new diff, I do actually see a lot of those affected tests using `{{.*}}` in various places already, so will just update them the way you suggested under the assumption that this is indeed a difference in test styles. If it turns out you actually like the current version after all, thanks to Git, it's easy to go back. :)